### PR TITLE
Polish standard trainers' catalog (Just rename some variables)

### LINKED ISSUE #2680: Make trainer input arguments' names consistent
Just by looking at [StandardLearnersCatalog.cs](https://github.com/dotnet/machinelearning/blob/1f90f50f813ecd152e46e92deb4a7827c7f24e31/src/Microsoft.ML.StandardLearners/StandardLearnersCatalog.cs) I found a lot of inconsistent naming for the binary classification trainer arguments. 

Below are the many naming variations that need to be conflated (both in extension method signatures and the Options properties):
* labelColumnName, labelColumn
* featureColumnName, featureColumn
* weightColumnName, weights
* initLearningRate, learningRate
* l2Weight, l2Const, l2RegularizerWeight
* l1Threshold, l1Weight
* numIterations, maxIterations
* lossFunction, loss
* numIterations, NumberOfIterations

I haven't checked yet, but it's probably the case for other trainers. There's probably more of these inconsistencies.

### PR DESCRIPTION
**Pull request by wschin**
To fix #2680, we rename several parameters and improve some doc strings.

### CODE REVIEW

File: src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs [hunk 1]
```diff
@@ -51,10 +51,10 @@ public abstract class OptionsBase : TrainerInputBaseWithWeight
             /// Number of previous iterations to remember for estimate of Hessian.
             /// </summary>
             [Argument(ArgumentType.AtMostOnce, HelpText = "Memory size for L-BFGS. Low=faster, less accurate",
-                ShortName = "m, MemorySize", SortOrder = 50)]
+                Name = "HistorySize", ShortName = "m, MemorySize", SortOrder = 50)]
             [TGUI(Description = "Memory size for L-BFGS", SuggestedSweeps = "5,20,50")]
             [TlcModule.SweepableDiscreteParamAttribute("MemorySize", new object[] { 5, 20, 50 })]
-            public int HistorySize = Defaults.HistorySize;
+            public int HistoryLength = Defaults.HistoryLength;
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs]: >HistoryLength [](start = 23, length = 13)

[https://github.com/dotnet/machinelearning/issues/2882](https://github.com/dotnet/machinelearning/issues/2882)
I though we agreed on `HistorySize`
Why this sudden change?

Comment (wschin) [reply src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs]: I didn't in that PR (sorry for missing the associated issue). `Size` usually means number of bytes to developers.

File: src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs [hunk 2]
```diff
@@ -25,8 +25,8 @@ public static class StandardTrainersCatalog
         /// <param name="featureColumnName">The features, or independent variables.</param>
         /// <param name="exampleWeightColumnName">The name of the example weight column (optional).</param>
         /// <param name="numberOfIterations">The maximum number of passes through the training dataset; set to 1 to simulate online learning.</param>
-        /// <param name="initialLearningRate">The initial <a href="tmpurl_lr">learning rate</a> used by SGD.</param>
-        /// <param name="l2Regularization">The L2 weight for <a href='tmpurl_regularization'>regularization</a>.</param>
+        /// <param name="learningRate">The learning rate. Large value may boost the training process but also leads to numerical instability.</param>
```
Comment (rogancarr) [inline src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs]: > boost the training process [](start = 73, length = 27)

"boost the training process" is unclear. Let's keep the `tmpurl` links in these. Shahab is adding these to all point to a common set of definitions for learning rate, L1, L2, etc.

File: src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs [hunk 3]
```diff
@@ -135,10 +135,10 @@ public static class StandardTrainersCatalog
         /// <param name="labelColumnName">The name of the label column.</param>
         /// <param name="featureColumnName">The name of the feature column.</param>
         /// <param name="exampleWeightColumnName">The name of the example weight column (optional).</param>
-        /// <param name="l2Regularization">The L2 <a href='tmpurl_regularization'>regularization</a> hyperparameter.</param>
-        /// <param name="l1Threshold">The L1 <a href='tmpurl_regularization'>regularization</a> hyperparameter. Higher values will tend to lead to more sparse model.</param>
```
Comment (rogancarr) [inline src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs]: ><a href='tmpurl_regularization'> [](start = 45, length = 32)

Just another note that we will want to point to a standard doc for L1 etc. and not put details here.

Comment (rogancarr) [reply src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs]: Although `Threshold` => `Regularization` is great!

---
In reply to: [267521149](https://github.com/dotnet/machinelearning/pull/3029#discussion_r267521149) [](ancestors = 267521149)

File: src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs [hunk 4]
```diff
@@ -507,7 +507,7 @@ public IClassificationLoss CreateComponent(IHostEnvironment env)
         /// <param name="enforceNonNegativity">Enforce non-negative weights.</param>
         /// <param name="l1Regularization">Weight of L1 regularization term.</param>
         /// <param name="l2Regularization">Weight of L2 regularization term.</param>
-        /// <param name="historySize">Memory size for <see cref="Trainers.LogisticRegressionBinaryTrainer"/>. Low=faster, less accurate.</param>
+        /// <param name="historyLength">Memory size for <see cref="Trainers.LogisticRegressionBinaryTrainer"/>. Low=faster, less accurate.</param>
```
Comment (rogancarr) [inline src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs]: >"historyLength [](start = 24, length = 14)

Let's keep `HistorySize`, as that's the common nomenclature for this parameter.

File: src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs (hunk 1 above)
Comment (rogancarr) [reply src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs]: `HistorySize` is standard nomenclature in the optimization field, so I'd prefer to keep it so. I don't think `Size` necessarily means size of memory (but I could be wrong :)).

---
In reply to: [267485468](https://github.com/dotnet/machinelearning/pull/3029#discussion_r267485468) [](ancestors = 267485468)

File: src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs (hunk 2 above)
Comment (rogancarr) [inline src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs]: >learningRate [](start = 25, length = 12)

Actually, this does mean the initial learning rate. The trainer dynamically rescales the learning rate, so it could be useful to keep calling it the initial learning rate.

File: test/Microsoft.ML.StaticPipelineTesting/Training.cs [hunk 5]
```diff
@@ -1132,7 +1132,7 @@ public void HogwildSGDSupportVectorMachine()
 
             var est = reader.MakeNewEstimator()
                 .Append(r => (r.label, preds: catalog.Trainers.StochasticGradientDescentNonCalibratedClassificationTrainer(r.label, r.features, null,
-                    new SgdNonCalibratedTrainer.Options { L2Regularization = 0, NumberOfThreads = 1, Loss = new HingeLoss()},
+                    new SgdNonCalibratedTrainer.Options { L2Regularization = 0, NumberOfThreads = 1, LossFunction = new HingeLoss()},
```
Comment (rogancarr) [inline test/Microsoft.ML.StaticPipelineTesting/Training.cs]: >LossFunction [](start = 101, length = 12)

Good!

File: test/Microsoft.ML.Tests/TrainerEstimators/MetalinearEstimators.cs [hunk 6]
```diff
@@ -25,7 +25,7 @@ public void OVAWithAllConstructorArgs()
                 new AveragedPerceptronTrainer.Options { Shuffle = true });
 
             var ova = ML.MulticlassClassification.Trainers.OneVersusAll(averagePerceptron, imputeMissingLabelsAsNegative: true,
-                calibrator: calibrator, maxCalibrationExamples: 10000, useProbabilities: true);
+                calibrator: calibrator, maximumCalibrationExampleCount: 10000, useProbabilities: true);
```
Comment (rogancarr) [inline test/Microsoft.ML.Tests/TrainerEstimators/MetalinearEstimators.cs]: >maximumCalibrationExampleCount [](start = 40, length = 30)

Do we really need to expand out the names? I'm on the fence. Good for clarity, bad for length.

Comment (wschin) [reply test/Microsoft.ML.Tests/TrainerEstimators/MetalinearEstimators.cs]: Yes, we have many similar names in tree trainers.

File: src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs (hunk 1 above)
Comment (wschin) [reply src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs]: How would users describe the `size` of history? Length? Size? I like `Length` a bit more but I am ok with the other choice too, so let's do `Size`.

File: src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs (hunk 2 above)
Comment (wschin) [reply src/Microsoft.ML.StandardTrainers/StandardTrainersCatalog.cs]: I took a short search for the uses of learning rates in other packages. It looks like no one is using initial learning rate.
TF: https://www.tensorflow.org/api_docs/python/tf/train/AdamOptimizer
Scikit-learn: https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDClassifier.html#sklearn.linear_model.SGDClassifier

File: src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs (hunk 1 above)
Comment (Ivanidzo4ka) [reply src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs]: We already had that discussion and we come up with `HistorySize`.
This is gonna be third time we rename this parameter, and I'm against another renaming for sake of renaming.

---
In reply to: [267585933](https://github.com/dotnet/machinelearning/pull/3029#discussion_r267585933) [](ancestors = 267585933)

Comment (wschin) [reply src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs]: Yes, so `Size` is used.

Comment (wschin) [reply src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs]: Looks like I miss some `HistoryLength`s. I am fixing them.

File: src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs [hunk 7]
```diff
@@ -51,7 +51,7 @@ public abstract class OptionsBase : TrainerInputBaseWithWeight
             /// Number of previous iterations to remember for estimate of Hessian.
             /// </summary>
             [Argument(ArgumentType.AtMostOnce, HelpText = "Memory size for L-BFGS. Low=faster, less accurate",
-                ShortName = "m, MemorySize", SortOrder = 50)]
+                Name = "HistorySize", ShortName = "m, MemorySize", SortOrder = 50)]
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/LbfgsPredictorBase.cs]: >HistorySize [](start = 24, length = 11)

not necessary change
